common/src/main/java/com/wynntils/services/itemfilter/ItemFilterService.java

```diff
@@ -0,0 +1,186 @@
+                String inputString = token.substring(token.indexOf(':') + 1);
+                ErrorOr<? extends ItemFilter> filterOrError = Services.ItemFilter.getFilter(filterString);
+
+                // If the filter does not exist, mark the token as ignored and continue to the next token
+                if (filterOrError.hasError()) {
+                    ignoredCharIndices.addAll(IntStream.rangeClosed(tokenStartIndex, tokenStartIndex + token.length())
+                            .boxed()
+                            .toList());
+                    errors.add(filterOrError.getError());
+                    continue;
+                }
+
+                // The filter exists, highlight the keyword...
+                validFilterCharIndices.addAll(
+                        IntStream.rangeClosed(tokenStartIndex, tokenStartIndex + filterString.length())
+                                .boxed()
+                                .toList());
+
+                // ...and try to create an instance for this filter only if the inputString is not empty,
+                // because filtering with an empty value is pointless and we don't want to show the error
+                // the ItemFilter might return because of that.
+                // We still want to highlight the filter keyword though, that's why we handle the empty input here.
+                if (inputString.isEmpty()) continue;
+
+                ErrorOr<? extends ItemFilterInstance> filterInstanceOrError =
+                        filterOrError.getValue().createInstance(inputString);
+
+                // If the inputString is invalid, mark the value as ignored and continue to the next token
+                if (filterInstanceOrError.hasError()) {
+                    ignoredCharIndices.addAll(IntStream.rangeClosed(
+                                    tokenStartIndex + filterString.length() + 1, tokenStartIndex + token.length())
+                            .boxed()
+                            .toList());
+                    errors.add(filterInstanceOrError.getError());
+                    continue;
+                }
+
+                // The inputString is valid, add the filter to the list
+                itemFilters.add(filterInstanceOrError.getValue());
+            } else if (!token.isEmpty()) {
+                // The token is not a filter, add it to the list of plain text tokens
+                plainTextTokens.add(token);
+            }
+        }
+
+        return new ItemSearchQuery(
+                queryString, itemFilters, ignoredCharIndices, validFilterCharIndices, errors, plainTextTokens);
+    }
+
+    /**
+     * Checks if the given item matches the search query. The item must match all filters and contain all plain text
+     * tokens. Therefore, if there are no plain text tokens, and no filters, this would be considered a match.
+     * <br>
+     * If the item is not a WynnItem, this method always returns false.
```

Comment:
Commendable with the javadoc! We have been quite sloppy at this. For most methods it is better to skip it, but the public methods of services, models and other components, we should really have them.